kaggle_environments/agent.py

```diff
@@ -12,23 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
 import os
 import requests
-import json
-from multiprocessing import Manager, Process
+import sys
+from io import StringIO
 from time import time
-import uuid
-from .errors import DeadlineExceeded
-from .utils import get_exec, has, is_url, read_file, structify
+from urllib.parse import urlparse
+from .errors import DeadlineExceeded, InvalidArgument
+from .utils import read_file, structify
+
+
+def is_url(url):
```

Comment:
is_url and get_last_callable are moved here from utils as this is the only callsite.